store/storetest/post_store.go

```diff
@@ -1972,3 +1973,56 @@ func testPostStoreGetRepliesForExport(t *testing.T, ss store.Store) {
 	assert.Equal(t, reply1.Username, u1.Username)
 
 }
+
+func testPostStoreGetDirectPostParentsForExportAfter(t *testing.T, ss store.Store) {
+	teamId := model.NewId()
+
+	o1 := model.Channel{}
+	o1.TeamId = teamId
+	o1.DisplayName = "Name"
+	o1.Name = "zz" + model.NewId() + "b"
+	o1.Type = model.CHANNEL_DIRECT
+
+	u1 := &model.User{}
+	u1.Email = MakeEmail()
+	u1.Nickname = model.NewId()
+	store.Must(ss.User().Save(u1))
+	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: model.NewId(), UserId: u1.Id}, -1))
+
+	u2 := &model.User{}
+	u2.Email = MakeEmail()
+	u2.Nickname = model.NewId()
+	store.Must(ss.User().Save(u2))
+	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: model.NewId(), UserId: u2.Id}, -1))
+
+	m1 := model.ChannelMember{}
+	m1.ChannelId = o1.Id
+	m1.UserId = u1.Id
+	m1.NotifyProps = model.GetDefaultChannelNotifyProps()
+
+	m2 := model.ChannelMember{}
+	m2.ChannelId = o1.Id
+	m2.UserId = u2.Id
+	m2.NotifyProps = model.GetDefaultChannelNotifyProps()
+
+	<-ss.Channel().SaveDirectChannel(&o1, &m1, &m2)
+
+	p1 := &model.Post{}
+	p1.ChannelId = o1.Id
+	p1.UserId = u1.Id
+	p1.Message = "zz" + model.NewId() + "AAAAAAAAAAA"
+	p1.CreateAt = 1000
+	p1 = (<-ss.Post().Save(p1)).Data.(*model.Post)
+
+	r1 := <-ss.Post().GetDirectPostParentsForExportAfter(10000, strings.Repeat("0", 26))
```

Comment:
Can we extend the unit tests here to:
* create 100+ direct messages, call with limit `100` and verify batches?
* tests for group messages
* verify that deleted channels are excluded
* verify that deleted users are excluded
* verify that public and private channels are excluded

Reply:
Nice! Indeed we can.

Reply:
While addressing this one, something occurred to me. Here's an example:
DM Channel containing two people. One person leaves and get deleted. If the code in `SqlChannelStore` retrieves only members with `DeleteAt` 0 for populating the `ChannelMembers` array, the moment this result is returned to the exporter, the exporter will drop the channel since it will treat it as a "me channel" or a channel containing a single person. So there's filtering in happening in two different layers here. How should we proceed?
Dropping DM and GM `Channels` and `Posts` is easy because the "main" query already filters them and it makes sense.